### PR TITLE
Move doc site to new Hyperledger template

### PR DESCRIPTION
**Pull request by nguyer**
This PR updates the doc site to use the new Hyperledger recommended framework using `mkdocs` and GitHub Pages. It uses a plugin called `mike` to manage the versions
- GitHub workflows have been updated to use this new build as well
-  All of this should fix the problem we always had when someone released a hotfix release for an older version of FireFly it  messed up the docs site in the past, because the concept of `latest` was ambiguous
- I've backported the docs for one previous release v1.2.2 into the new framework so it shows up in the version list. If we feel like we need more than that, we can do additional releases, but it's a bit of manual work

New version of the doc site is live at: https://hyperledger.github.io/firefly/latest/

### CODE REVIEW

File: internal/events/websockets/websockets_test.go [hunk 1]
```diff
@@ -1220,3 +1220,10 @@ func TestHandleStartWrongNamespace(t *testing.T) {
 	assert.Error(t, err)
 	assert.Regexp(t, "FF10462", err)
 }
+
+func TestIsBoolQuerySet(t *testing.T) {
+	assert.True(t, isBoolQuerySet(url.Values{"batch": []string{"true"}}, "batch"))
+	assert.True(t, isBoolQuerySet(url.Values{"batch": []string{}}, "batch"))
+	assert.False(t, isBoolQuerySet(url.Values{}, "batch"))
+	assert.False(t, isBoolQuerySet(url.Values{"batch": []string{"false"}}, "batch"))
+}
```
Comment (EnriqueL8) [inline internal/events/websockets/websockets_test.go:1229]: Did you want to add this in this PR?

**Review by EnriqueL8 — COMMENTED** [review]
Awesome stuff!  One small comment


Might be also worth adding a section to the README of the docs of how the documentation is versioned based on tags and how the head/latest works?

File: doc-site/.github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<!--  Thanks for sending a pull request! -->
+
+## Proposed changes
+
+Please include a summary of the changes here and why we need those changes. And also let us know which issue is fixed.
```
Comment (EnriqueL8) [inline doc-site/.github/pull_request_template.md:5]: Awesome!

**Comment by nguyer** [comment]
> Awesome stuff!  One small comment
> 
> Might be also worth adding a section to the README of the docs of how the documentation is versioned based on tags and how the head/latest works?

Yeah it is also on my mind that I need to update the docs on how to update the docs. We have a page in the Contributors' guide on that but it is no longer accurate now. Will push one more commit for that.

**Review by EnriqueL8 — APPROVED** [review]
Looks great to me! Thanks for providing the preview to help review this 😃

**Review by awrichar — APPROVED** [review]
LGTM! Thanks for tackling this.

**Pull request merged**